### PR TITLE
Add type definitions for zetapush-js

### PR DESCRIPTION
**Pull request by ghoullier**
This PullRequest adds type definitions for the zetapush-js NPM module
The original author of these type definitions is Grégory Houllier https://github.com/ghoullier
The author of zetapush-js is zetapush https://github.com/zetapush/zetapush-js

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

File: types/zetapush-js/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+export class SmartClient extends Client {
+  getCredentials(): any;
+  getSession(): any;
+  hasCredentials(): boolean;
+  isStronglyAuthenticated(session?: any): boolean;
+  isWeaklyAuthenticated(session?: any): boolean;
+  setCredentials(credentials: any): void;
+}
+
+export const services: Helper.Services;
```
Comment (rbuckton) [inline types/zetapush-js/index.d.ts]: After reviewing the documentation, I would also recommend you add an `export as namespace ZetaPush;` to the file, for those using it via CDN (https://github.com/Microsoft/TypeScript-Handbook/blob/master/pages/Modules.md#umd-modules).

**Review by rbuckton — CHANGES_REQUESTED** [review]

**Comment by ghoullier** [comment]
@rbuckton I've update my test and definition accorting to your review

File: types/zetapush-js/index.d.ts (hunk 1 above)
Comment (ghoullier) [inline types/zetapush-js/index.d.ts]: I'v added to following statement
```typescript
export as namespace ZetaPush;
```

**Comment by yuit** [comment]
ping @rbuckton

**Comment by andy-ms** [comment]
Thanks!

**Pull request merged**